spec/index.bs

```diff
@@ -2748,82 +2800,88 @@ A {{GPUBindGroup}} object has the following internal slots:
                             For each {{GPUBindGroupEntry}} |bindingDescriptor| in
                                 |descriptor|.{{GPUBindGroupDescriptor/entries}}:
                                 - Let |resource| be |bindingDescriptor|.{{GPUBindGroupEntry/resource}}.
+                                - Let |bindingType| be |layoutBinding|.{{GPUBindGroupLayoutEntry/type}}.
                                 - There is exactly one {{GPUBindGroupLayoutEntry}} |layoutBinding|
                                     in |descriptor|.{{GPUBindGroupDescriptor/layout}}.{{GPUBindGroupLayoutDescriptor/entries}}
                                     such that |layoutBinding|.{{GPUBindGroupLayoutEntry/binding}} equals to
                                     |bindingDescriptor|.{{GPUBindGroupEntry/binding}}.
 
-                                - If |layoutBinding|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"sampler"}}:
-                                    - |resource| is [$valid to use with$] |this|.
-                                    - |resource|.{{GPUSampler/[[compareEnable]]}} is `false`.
+                                - If the [=binding resource type=] for |bindingType| is
+                                    <dl class="switch">
+                                        : {{GPUSampler}}
+                                        ::
+                                            - |resource| is a {{GPUSampler}}.
+                                            - |resource| is [$valid to use with$] |this|.
+                                            - If |bindingType| is
```

Comment:
I was actually thinking that the giant switch would be on the binding type, not `[=binding resource type=]`

Reply:
That's doable, but it would lead to a quite a bit of repeated validation steps between types.

Reply:
I thought we can have intersected ranges in the switch, i.e. 
```
switch {
 A| B | C => ...,
 A => ...,
 B => ...,
}
```

Reply:
Hm... I don't find that terribly intuitive to read. I tend to think of switch statements as executing the first block that satisfies the condition and then moving on. (With the exception of fallthrough, but let's not speak of such evils here.)

I'd like to hear what some of the other editors think on that matter. If it's a pattern that you'd all like to see I'm happy to run with it.

Reply:
On the one hand, it makes sense to think of switches that way, and it's the least ambiguous. On the other, it's clear from context in this case that we wouldn't write overlapping switch cases unless it was intentional. But there are some situations where that's not clear, e.g.
```
switch {
 A && B => ...,
 A => ...,
}
```
so it's probably best to avoid it.

I haven't looked closely at the text but perhaps this can be phrased well as multiple switches?

```
switch {
 A | B | C => ...,
}
switch {
 A => ...,
 B => ...,
}

Reply:
Now that I've looked at how this renders, I actually quite like the current nested switch.